### PR TITLE
Add basic test for jsonCompiler

### PR DESCRIPTION
**Pull request by axic**
(Part of #556)

### CODE REVIEW

File: test/libsolidity/JSONCompiler.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+{
+	char const* input = R"(
+	{
+		"sources": {
+			"fileA": "contract A { }"
+		}
+	}
+	)";
+	Json::Value result = compile(input);
+//	BOOST_CHECK(containsAtMostWarnings(result));
```
Comment (axic) [inline test/libsolidity/JSONCompiler.cpp]: I don't think we need to check for errors. Let's assume this just works.

File: test/Metadata.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+#include <string>
+
+namespace dev
+{
+namespace test
+{
+
+std::string bytecodeSansMetadata(std::string const& _bytecode);
```
Comment (chriseth) [inline test/Metadata.h]: Please document (both).

File: test/libsolidity/JSONCompiler.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+{
+namespace test
+{
+
+namespace
+{
+
+Json::Value compile(string const& _input)
+{
+	string output(compileJSONMulti(_input.c_str(), false));
```
Comment (chriseth) [inline test/libsolidity/JSONCompiler.cpp]: We run all tests once with optimizer disabled and once with optimizer enabled. Can we do the same here? I think it should be a function call in the Options class.

File: test/libsolidity/JSONCompiler.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+{
+	char const* input = R"(
+	{
+		"sources": {
+			"fileA": "contract A { }"
+		}
+	}
+	)";
+	Json::Value result = compile(input);
+	std::cout << result.toStyledString() << endl;
```
Comment (chriseth) [inline test/libsolidity/JSONCompiler.cpp]: Please remove.

**Review by chriseth — APPROVED** [review]

File: test/libsolidity/JSONCompiler.cpp (hunk 3 above)
Comment (axic) [reply test/libsolidity/JSONCompiler.cpp]: Good idea, though the bytecode would be different.

**Pull request merged**